R/annotated_data.R

```diff
@@ -0,0 +1,1350 @@
+
+setMethod(
+  f = "annotate"
+  , signature = "data.frame"
+  , definition = function(object){
+    # ----
+    # optional=TRUE suppresses changing colnames
+    as.data.frame(lapply(X = object, FUN = annotate), stringsAsFactors = FALSE, optional = TRUE)
+  }
+)
+
+#' @rdname annotate
+#' @keywords internal
+
+setMethod(
+  f = "annotate"
+  , signature = "vector"
+  , definition = function(object){
+    if(is.null(names(object))){
+      as(object, paste0("annotated_", class(object)))
+    } else {
+      as(object, paste0("annotated_named_", class(object)))
+    }
+  }
+)
+
+#' @rdname annotate
+#' @keywords internal
+
+setMethod(
+  f = "annotate"
+  , signature = "annotated_vector"
+  , definition = function(object){
+    object
+  }
+)
+
+
+
+
+
+
+#' Set Levels of an Annotated Factor
+#'
+#' This method is as wrapper around \code{\link{levels}} and should do the same.
+#' Annotations are preserved.
+#'
+#' @param x An object of \code{annotated_factor-class}.
+#' @param value Character. The valid value for \code{levels(x)}.
+#'
+#' @keywords internal
+
+setMethod(
+  "levels<-"
+  , "annotated_factor"
+  , definition = function(x, value){
+    fac <- factor(x@levels[x@.Data], levels = x@levels)
+    levels(fac) <- value
+    new("annotated_factor", fac, levels = levels(fac), annotation = x@annotation, label = x@label)
+  }
+)
+
+
+
+#' Replicate Elements of Annotated Vectors
+#'
+#' Methods for annotated vectors to replicate values, calls \code{\link{rep}}
+#' and preserves annotations.
+#'
+#' @param x An object of class \code{annotated_vector}.
+#' @param ... Further arguments that can be passed to\code{\link{rep}}.
+#'
+#'
+#' @rdname rep
+#' @keywords internal
+
+setMethod(
+  "rep"
+  , signature = "annotated_vector"
+  , definition = function(x, ...){
+    new(
+      class(x)
+      , .Data = rep(x@.Data, ...)
+      , annotation = x@annotation
+      , label = x@label
+    )
+  }
+)
+
+#' @rdname rep
+#' @keywords internal
+
+setMethod(
+  "rep"
+  , signature = "annotated_factor"
+  , definition = function(x, ...){
+    new(
+      "annotated_factor"
+      , .Data = rep(x@.Data, ...)
+      , annotation = x@annotation
+      , levels = x@levels
+      , label = x@label
+    )
+  }
+)
+
+
+
+
+#' The Names of an Annotated Vector
+#'
+#' Methods to set the names of an annotated vector.
+#' If names are specified for an unnamed annotated_vector, its class is changed
+#' to the corresponding class from annotated_named_vector-superclass.
+#' @param x An annotated vector.
+#' @param value A character vector of the same length as x, or NULL.
+#' @details
+#'   See \code{\link{names}} for details. If the names of an object of an
+#'   annotated_named_vector are removed by setting \code{names(x) <- NULL}, it
+#'   is coerced to the corresponding class from annotated_vector-superclasses
+#'   (i.e., an annotated vector without a names slot).
+#' @rdname names-set-annotated_vector-ANY-method
+#' @keywords internal
+
+setMethod(
```

Comment:
Could these two methods be unified by using a more sophisticated regex (e.g. `annotated_(?!named)`)?

Reply:
I would love to play around with this when we already merged (and, subsequently, might have more meaningful tests), so that I can check the consistency of both approaches.

Reply:
Fair enough.